sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/testresource/StargateTestResource.java

```diff
@@ -0,0 +1,279 @@
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+/**
+ * Quarkus test resource that starts Cassandra/DSE and Stargate Coordinator using test containers.
+ *
+ * <p>Should be used in the integration tests, that should be explicitly annotated with {@link
+ * IntegrationTestProfile} class: <code>@TestProfile(IntegrationTestProfile.class)</code>
+ *
+ * <p>When run from IDE, by default it uses container versions specified in {@link Defaults}. If you
+ * wish to run locally with different Cassandra version or the DSE, please set up following system
+ * properties:
+ *
+ * <ol>
+ *   <li><code>testing.containers.cassandra-image</code>
+ *   <li><code>testing.containers.stargate-image</code>
+ *   <li><code>testing.containers.cluster-version</code>
+ *   <li><code>testing.containers.cluster-dse</code>
+ * </ol>
+ *
+ * <p>Note that this resource fetches the auth token and sets it via the properties, using the
+ * {@link io.stargate.sgv2.docsapi.api.common.token.impl.FixedTokenResolver}.
+ */
+public class StargateTestResource
+    implements QuarkusTestResourceLifecycleManager, DevServicesContext.ContextAware {
+
+  /** Set of defaults for the integration tests, usually used when running from IDE. */
+  interface Defaults {
+
+    String CASSANDRA_IMAGE = "cassandra";
+    String CASSANDRA_IMAGE_TAG = "4.0.3";
+
+    String STARGATE_IMAGE = "stargateio/coordinator-4_0";
+    String STARGATE_IMAGE_TAG = "latest";
+
+    String CLUSTER_NAME = "int-test-cluster";
+    String CLUSTER_VERSION = "4.0";
+
+    String CLUSTER_DSE = null;
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(StargateTestResource.class);
+
+  private boolean enabled;
+
+  private Optional<String> containerNetworkId;
+
+  private Network network;
+
+  private GenericContainer<?> cassandraContainer;
+
+  private GenericContainer<?> stargateContainer;
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p><i>Note: container network ID will be present if resource is used
+   * with @QuarkusIntegrationTest</i>
+   */
+  @Override
+  public void setIntegrationTestContext(DevServicesContext context) {
+    containerNetworkId = context.containerNetworkId();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p><i>Note: only enable if IntegrationTestProfile is active profile</i>
+   */
+  @Override
+  public void setContext(Context context) {
+    String testProfile = context.testProfile();
+    String expectedProfile = IntegrationTestProfile.class.getName();
+    this.enabled = Objects.equals(testProfile, expectedProfile);
```

Comment:
Yes so this is to only enable the `StargateTestResource` in the integration tests.. For whatever reason, the Quarkus seems to start the `StargateTestResource` even if no "unit" test is annotated with it.. Seems like some kind of bug related to class path scanning.. 

I was hoping that this would be fixed in the `2.9.1`, but seems not.. 

So for now I am keeping a dedicated integration profile as a marker on when the resource should be started.. I opened a ticket at https://github.com/quarkusio/quarkus/issues/25782

Reply:
Fixed in https://github.com/stargate/stargate/pull/1825/commits/971d52ae99f4d2402270817f9e905ec5cfabb2a1 :guitar: